pallets/staking-rewards/src/lib.rs

```diff
@@ -427,20 +428,17 @@ fn pay_rent_rewards_out<T: Config>(maybe_active_era_info: Option<ActiveEraInfo>)
         return;
     }
 
-    let funds: u128 = pallet::Pallet::<T>::rent_pool_free_balance().unique_saturated_into();
+    let funds: u128 = pallet::Pallet::<T>::rent_pool_balance().unique_saturated_into();
     for (account_id, points) in reward_points.individual {
         let payout = funds.saturating_mul(u128::from(points)) / total;
         if payout > 0 {
-            let result = CurrencyOf::<T>::transfer(
+            CurrencyOf::<T>::transfer(
                 &pallet::Pallet::<T>::rent_pool_account_id(),
                 &account_id,
                 payout.unique_saturated_into(),
                 ExistenceRequirement::KeepAlive,
-            );
-
-            if result.is_err() {
-                log::debug!("Failed to transfer rent reward, account_id = {account_id:#?}, points = {points}, payout = {payout}");
-            }
+            )
+            .unwrap_or_else(|e| unreachable!("Failed to transfer rent reward: {e:?}; account_id = {account_id:#?}, points = {points}, payout = {payout}"));
```

Comment:
could theoretically validators balance not exist?

Reply:
Just wondering if we may want to log::error instead of some crucial unreachable. More recovery cases, especially bcz this code funds-sensitive, but we can always restore some data and grant validator properly if something went wrong. If I'm not mistaken same we do for staking rewards and rent pool on burns, isn't it?